### PR TITLE
Adding gitignore file and support for go modules [fix #30]

### LINKED ISSUE #30: Add support for go.mod versioning
The standard "clone > build" process is being impacted by the Go modules file.

Evidence:
`
$ ./rebuild.sh 
`
Is resulting in

`Checking for Go
Getting dependencies
go: go.mod file not found in current directory or any parent directory.
	'go get' is no longer supported outside a module.
	To build and install a command, use 'go install' with a version,
	like 'go install example.com/cmd@latest'
	For more information, see https://golang.org/doc/go-get-install-deprecation
	or run 'go help get' or 'go help install'.
go: go.mod file not found in current directory or any parent directory.
	'go get' is no longer supported outside a module.
	To build and install a command, use 'go install' with a version,
	like 'go install example.com/cmd@latest'
	For more information, see https://golang.org/doc/go-get-install-deprecation
	or run 'go help get' or 'go help install'.
Building asset creator
mkassets.go:30:2: no required module provides package gopkg.in/yaml.v2: go.mod file not found in current directory or any parent directory; see 'go help modules'`

Even though it's possible to manually set the var GO111MODULE=off. It would be good to add the go.mod file to allow for a more straightforward build process.

### PR DESCRIPTION
**Pull request by zamariola**
I updated the go version to 1.19 but it could be 1.20. I went with the oldest one.
Added some gitignore entries as well.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#binaries
+mkassets
+proxima5
+
+#generated assets
```
Comment (gdamore) [inline .gitignore]: While these assets are generated, the generated results are intended to be under git.  Not everyone should need to regenerate them every time.  Please don't include them here.

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module github.com/gdamore/proxima5
```
Comment (gdamore) [inline go.mod:1]: Lol.  This program predates the invention of go modules. :-)

**Review by gdamore — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (zamariola) [reply .gitignore]: It makes sense! I'll add them as well

**Comment by zamariola** [comment]
it's done! I removed them from the ignore and pushed the newly locally generated ones! Thanks!

**Review by gdamore — APPROVED** [review]

**Pull request merged**